### ISSUE TITLE
[New Resource]: OpenSearch Ingestion Pipeline

### ISSUE THREAD

**Issue by farhanangullia** [hashicorp/terraform-provider-aws#31364]
### Description

Amazon OpenSearch Ingestion is a fully managed, serverless data collector that delivers real-time log, metric, and trace data to Amazon OpenSearch Service domains and OpenSearch Serverless collections.

### Requested Resource(s) and/or Data Source(s)

* aws_osis_pipeline

### Potential Terraform Configuration

```terraform
resource "aws_osis_pipeline" "example" {
  name                        = "example"
  max_units                   = 5
  min_units                   = 1
  pipeline_configuration_body = file("${path.module}/pipeline.yaml")
  log_publishing_options {
    cloudwatch_log_destination {
      log_group = "/aws/OpenSearchService/IngestionService/my-pipeline"
    }
    is_logging_enabled = true
  }
}
```

### References

https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.44.262/service/osis

### Would you like to implement a fix?

Yes

**Comment by lawliet89** [comment]
[`awscc_osis_pipeline`](https://registry.terraform.io/providers/hashicorp/awscc/latest/docs/resources/osis_pipeline) is available for use in the meantime.

**Comment by komgritz** [comment]
I am waiting for available on hashicorp/aws provider.

**Comment by DanielViglione** [comment]
I do not recommend the “awscc_osis_pipeline” resource. The awscc provider is pretty inflexible. If you use awscc_osis_pipeline, you will feel compelled to use awscc_iam_role_policy and awscc_iam_role, for example. awscc_iam_role doesn’t work like aws_iam_role with regards to tags, the outputs are named differently and worst of all the awscc_iam_role_policy only accepts a policy document as a map of strings, so you cannot pass it a json string, which makes it unsuitable for workloads where you want to create reusable modules and pass policies as json strings.

**Comment by HaroonSaid** [comment]
Any progress?

**Comment by edmondsiu0** [comment]
I echo @DanielViglione's opinion on not recommending `awscc_osis_pipeline` but for reason of access control / security. 

Access control to AWS resources through the AWS Cloud Control API is very crude (pun intended) - while you can restrict actions based on Create / Read / Update / Delete or List, this access is not granular and is given to all resources within the AWS account, AWS CC does not currently support resource-level permissions or conditions. 

Reference: https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/security.html

To me, to use `awscc_` resources is the same as deploying with the root AWS user.  
Basically throwing the principle of least privilege out of the window with this resource. 

We're in the process of evaluating the use of an alternative provider:  
https://registry.terraform.io/providers/opensearch-project/opensearch/latest/docs/resources/ingest_pipeline

**Comment by cameronattard** [comment]
> We're in the process of evaluating the use of an alternative provider: https://registry.terraform.io/providers/opensearch-project/opensearch/latest/docs/resources/ingest_pipeline

FYI that is for a totally separate thing from AWS OpenSearch Ingestion Pipelines - https://opensearch.org/docs/latest/ingest-pipelines/create-ingest/

**Comment by thunder-spb** [comment]
Any updates on this?

**Comment by praddc** [comment]
Just curious if there are updates on this. Thanks!

**Comment by farhanangullia** [comment]
hi folks, i'll be raising the PR for this tomorrow.

**Comment by kamalika-majumder** [comment]
Will this be included in the next release of terraform provider for aws ?
